.travis/install.sh

```diff
@@ -62,26 +62,22 @@ else
         pyenv global pypy-4.0.1
     fi
     if [[ "${OPENSSL}" == "0.9.8" ]]; then
-      # download, compile, and install if it's not already present via travis cache
-      # This is 0.9.8l rather than zh because we have some branches for handling
-      # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
-      # patches, but while that's in jenkins we don't get coverage data from it)
-      if [[ ! -f "$HOME/ossl-098l/bin/openssl" ]]; then
-        curl -O https://www.openssl.org/source/openssl-0.9.8l.tar.gz
-        tar zxf openssl-0.9.8l.tar.gz
-        cd openssl-0.9.8l
-        echo "OPENSSL_0.9.8L_CUSTOM {
-            global:
-              *;
-        };" > openssl.ld
-        ./config no-asm no-ssl2 -Wl,--version-script=openssl.ld -Wl,-Bsymbolic-functions -fPIC shared --prefix=$HOME/ossl-098l
-        make depend
-        make install
-      fi
-      export PATH="$HOME/ossl-098l/bin:$PATH"
-      export CFLAGS="-I$HOME/ossl-098l/include"
-      export LDFLAGS="-L$HOME/ossl-098l/lib"
-      export LD_LIBRARY_PATH="$HOME/ossl-098l/lib"
+        # We use 0.9.8l rather than zh because we have some branches for handling
+        # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
+        # patches, but while that's in jenkins we don't get coverage data from it).
+        OPENSSL_VERSION_NUMBER="0.9.8l"
+        OPENSSL_DIR="ossl-098l"
+    fi
+    # download, compile, and install if it's not already present via travis cache
+    if [ ! -z "$OPENSSL_DIR" ]; then
+        if [[ ! -f "$HOME/$OPENSSL_DIR/bin/openssl" ]]; then
+            curl -O https://www.openssl.org/source/openssl-$OPENSSL_VERSION_NUMBER.tar.gz
+            tar zxf openssl-$OPENSSL_VERSION_NUMBER.tar.gz
+            cd openssl-$OPENSSL_VERSION_NUMBER
+            ./config shared no-asm no-ssl2 -fPIC --prefix=$HOME/$OPENSSL_DIR
```

Comment:
:+1:

On Sun, Jan 3, 2016 at 11:18 AM, Paul Kehrer <notifications@github.com>
wrote:

> In .travis/install.sh
> <https://github.com/pyca/cryptography/pull/2629#discussion_r48694475>:
>
> > -      export CFLAGS="-I$HOME/ossl-098l/include"
> > -      export LDFLAGS="-L$HOME/ossl-098l/lib"
> > -      export LD_LIBRARY_PATH="$HOME/ossl-098l/lib"
> > +        # We use 0.9.8l rather than zh because we have some branches for handling
> > +        # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
> > +        # patches, but while that's in jenkins we don't get coverage data from it).
> > +        OPENSSL_VERSION_NUMBER="0.9.8l"
> > +        OPENSSL_DIR="ossl-098l"
> > +    fi
> > +    # download, compile, and install if it's not already present via travis cache
> > +    if [ ! -z "$OPENSSL_DIR" ]; then
> > +        if [[ ! -f "$HOME/$OPENSSL_DIR/bin/openssl" ]]; then
> > +            curl -O https://www.openssl.org/source/openssl-$OPENSSL_VERSION_NUMBER.tar.gz
> > +            tar zxf openssl-$OPENSSL_VERSION_NUMBER.tar.gz
> > +            cd openssl-$OPENSSL_VERSION_NUMBER
> > +            ./config shared no-asm no-ssl2 -fPIC --prefix=$HOME/$OPENSSL_DIR
>
> version-script wasn't actually doing anything useful unfortunately
> (verified with a whole lot of compiling over and over last night) and
> -Wl,-Bsymbolic-functions interacts poorly with -fPIC on OpenSSL 1.0.0t (for
> unclear reasons). We don't need it so I decided dropping it was the better
> part of valor.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2629/files#r48694475>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084